tools/istio-iptables/pkg/cmd/root.go

```diff
@@ -30,10 +32,13 @@ import (
 var rootCmd = &cobra.Command{
 	Use:  "istio-iptables",
 	Long: "Script responsible for setting up port forwarding for Istio sidecar.",
-	Run: func(cmd *cobra.Command, args []string) {
+	RunE: func(cmd *cobra.Command, args []string) error {
+		fmt.Println("lambdai")
 		config := constructConfig()
 		iptConfigurator := NewIptablesConfigurator(config)
```

Comment:
This is not correct. You can't setup iptables when Istio CNI is enabled, because 1) this container won't run with CAP_NET_ADMIN, and 2) trying to initialize iptables a 2nd time after Istio CNI has already done it will fail.

So your validator has to be in another command, not in the root command.

Reply:
The idea is to reuse the configurator defined here.

Ideally 
1. `cni` executes the iptable setup in the kubelet context 
2. `istio-init` executes the validation in the preinit pod
3. for non-cni, run both

WDYT?

Reply:
> 3. for non-cni, run both

Why? I don't understand why we would ever need to check an iptables that succeeded just before.

We would ever need to do one or the other. Which is why I recommended to do the validation in another command, to make that clear.

Reply:
I don't have strong opinion we must validate in non-cni.

But I'd like to ship iptables setup in one binary. Easy to maintain, and adding very little overhead to the binary size.
Isn't that a fact that istio-iptables is already running in both kubelet (for cni) context and istio-init(for non-cni)?

Reply:
> But I'd like to ship iptables setup in one binary.

I never asked for another binary.
I asked for another Cobra comamnd.

Reply:
@rlenglet Thanks! 
I add --skip-rule-apply and --skip-validation. 
See the summary